### PR TITLE
Changed verbiage in docs and error message for MailComposer

### PR DESCRIPTION
**Pull request by showcasecode**
# Why

Changed verbiage in docs and error message for MailComposer

# How

Changed verbiage in docs and error message for MailComposer

# Test Plan

Verbiage updates

### CODE REVIEW

File: packages/expo-mail-composer/ios/EXMailComposer/EXMailComposer.m [hunk 1]
```diff
@@ -35,7 +35,7 @@ - (dispatch_queue_t)methodQueue
                     rejecter:(UMPromiseRejectBlock)reject)
 {
   if (![MFMailComposeViewController canSendMail]) {
-    reject(@"E_COMPOSE_UNAVAILABLE", @"Mail services are not available.", nil);
+    reject(@"E_COMPOSE_UNAVAILABLE", @"Mail services are not available. Make sure you're signed into the Mail app", nil);
```
Comment (sjchmiela) [inline packages/expo-mail-composer/ios/EXMailComposer/EXMailComposer.m]: Good catch! Could you also make this change in `ios/versioned-react-native/ABI35_0_0/…/EXMailComposer.m`, so this change lands in SDK35 too (not only on `master` and SDK36+)?

File: docs/pages/versions/v34.0.0/sdk/mail-composer.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ import * as MailComposer from 'expo-mail-composer';
 
 ### `MailComposer.composeAsync(options)`
 
-Opens a mail modal for iOS and a mail app intent for Android and fills the fields with provided data.
+Opens a mail modal for iOS and a mail app intent for Android and fills the fields with provided data. On iOS you will need to be signed into the Mail app.
```
Comment (sjchmiela) [inline docs/pages/versions/v34.0.0/sdk/mail-composer.md]: For this change to be persisted in future SDKs, could you also make it in `docs/pages/versions/unversioned/sdk/mail-composer.md`?

**Review by sjchmiela — COMMENTED** [review]
Thank you for this contribution! Would you be so kind and make these changes in other files too?

**Comment by teckel12** [comment]
@sjchmiela @showcasecode @cruzach This really doesn't fix the problem, it just makes it more clear when developing what the problem is.  For the user, it's like the button doesn't work as it throws a warning and not an error which can be trapped.

File: packages/expo-mail-composer/ios/EXMailComposer/EXMailComposer.m (hunk 1 above)
Comment (teckel12) [reply packages/expo-mail-composer/ios/EXMailComposer/EXMailComposer.m:38]:
```suggestion
    reject(@"E_COMPOSE_UNAVAILABLE", @"Mail services are not available. Make sure you're signed into the Mail app", nil);
    resolve(@{ @"status": @"failed" });
```

**Comment by teckel12** [comment]
I solved it with this ugly code, but I believe there should be a more graceful method of doing this, for example adding another status result.

```
        async function tryMail() {
            try {
                await MailComposer.composeAsync({
                    subject,
                    body,
                    recipients: ['sales@yourcompany.com'],
                    isHtml: true
                })
                .then(({ status }) => {
                    analyticsEvent('Quote', status);
                });
            } catch {
                analyticsEvent('Quote', 'Failed/NoEmail');
                Alert.alert('Mail services not available', "\nMake sure you've setup email on your phone");
            }
        }

        tryMail();
```

**Pull request merged**